feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+    that maximizes performance while minimizing the number of features selected.
+
+    Attributes
+    ----------
+    selection_size: int
+        Number of content/feature to select.
+    seed: int
+        Random seed.
+    trials: int
+        Number of trials to run for random functions to estimate average coverage.
+    verbose: bool
+        Print intermediate results if True.
+    matrix: np.ndarray
+        Coverage matrix where each row corresponds to all predefined labels for each category and the columns
+        correspond to the content.
+    features: np.ndarray
+        Numeric featurization of content.
+    """
+
+    def __init__(self,  selection_size: int, seed: int = 123546, trials: int = 10, verbose: int = 0):
+        """ContentSelector
+
+       Creates a content selector object.
+
+       Parameters
+       ----------
+        selection_size: int
+            Maximum number of content to select.
+        seed: int, default=123456
+            Random seed.
+        trials: int, default=10
+            Number of trials to run for random functions to estimate average coverage.
+        verbose: int, default=0
+            Print intermediate results if 1.
+        """
+
+        # Validate arguments
+        self.featurization_method = None
+        self.cost_metric = None
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = verbose == 1
+
+        # Initialize class variables
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self,
+                              input_df: pd.DataFrame,
+                              categories: List[int], featurization_method: TextWiser,
+                              selection_size: int,
+                              optimization_method: str = "exact",
+                              cost_metric: str = "diverse",
+                              trials: int = 10) -> List:
+
+        """Run content selection algorithm.
+
+        Parameters
+        ----------
+        selection_size
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: List[int]
+            List of columns in data that contains categories/labels to be covered.
+
+        featurization_method: TextWiser
+
+        optimization_method: str, default="exact"
+            Optimization method used to perform content selection.
+            See TextBased in Selector.py for supported options definition
+
+        cost_metric: str, default = "diverse"
+
+        trials: int, default = 10
+
+        Returns
+        -------
+        List of indices in data that are selected.
+        """
+
+        # Process data
+        self.cost_metric = cost_metric
```

Comment:
when setting these variables, it is better to keep their ordering, in the same order as the function signature to avoid confusion